sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/execution_dispatcher.py

```diff
@@ -136,6 +145,22 @@ def _create_pipelined_execution_context(self, query_execution_info):
                                                                                         self._query,
                                                                                         self._options,
                                                                                         query_execution_info)
+        elif query_execution_info.has_hybrid_search_query_info():
```

Comment:
can thge exception factory be refactroed to share across sync and async?

Reply:
yep, done